### DISCUSSION TITLE
How do I pause the game but not the Dialog node?

### DISCUSSION THREAD

**Discussion by MadeySlime** [dialogic-godot/dialogic#1466] [Q&A] [answered]
Howdy,
I'm trying to use Dialogic for Godot 4 Stable (Using [Dialogic 2.0 Alpha 6)](https://github.com/coppolaemilio/dialogic/releases/tag/2.0-alpha-6) but I can't seem to pause the scene while using the Dialog.
I tried following [this tutorial](https://www.youtube.com/watch?v=gZ3h09B7Utc) but it seems outdated as it doesn't work for me.
Thanks in advance!

**Comment by Jowan-Spooner** [comment, marked as answer]
Hi @FirePrincessITA , the Dialogic.start() method currently doesn't return the layout scene, so you cannot change the pause mode. This will be fixed in the future. 
The turorial is indeed outdated (for dialogic 1), new documentation will come once dialogi 2 is getting closer to completion/stableness.

**Reply by MadeySlime** [reply]
Oh ok then, thanks!! I'll try to find another way then ^^

**Reply by Jowan-Spooner** [reply]
Okay I've also pushed an update, so that .start() now returns the node. This means you can now set the process_mode to ALWAYS. However you also have to set the process_mode of the dialogic singleton to always.

E.g.:
```gdscript
func _on_something_that_starts_dialog():
	Dialogic.start("res://test-project/example-timeline.dtl").process_mode = Node.PROCESS_MODE_ALWAYS
	Dialogic.process_mode = Node.PROCESS_MODE_ALWAYS
	Dialogic.timeline_ended.connect(get_tree().set('paused', false))
	get_tree().paused = true
```

**Comment by luminoucid** [comment]
Thanks @Jowan-Spooner !

In the Godot version 4.3.stable, dialogic 2.0-alpha-17 WIP, got the following error: 

```
Line 16:Cannot get return value of call to "set()" because it returns "void".
Line 16:Invalid argument for "connect()" function: argument 1 should be "Callable" but is "null".
```

I could fix the error by slightly modifying your code, adding **func():** into one line 4.

```
func _on_something_that_starts_dialog():
	Dialogic.start("res://test-project/example-timeline.dtl").process_mode = Node.PROCESS_MODE_ALWAYS
	Dialogic.process_mode = Node.PROCESS_MODE_ALWAYS
	Dialogic.timeline_ended.connect(func():get_tree().set('paused', false))
	get_tree().paused = true
```
